conclave-common/src/main/kotlin/com/r3/conclave/common/internal/CallInitiator.kt

```diff
@@ -0,0 +1,11 @@
+package com.r3.conclave.common.internal
+
+import java.nio.ByteBuffer
+
+interface CallInitiator<CALL_ID_TYPE> {
+    companion object {
+        private val EMPTY_BYTE_BUFFER: ByteBuffer = ByteBuffer.wrap(ByteArray(0)).asReadOnlyBuffer()
+    }
+
+    fun executeCall(callType: CALL_ID_TYPE, parameterBuffer: ByteBuffer = EMPTY_BYTE_BUFFER): ByteBuffer?
```

Comment:
It's just so that you can write `executeCall(<call type>)` without having to explicitly pass an empty byte buffer. It avoids a pointless object being created, that's all. I also considered making the parameter nullable. Thoughts?